internal/pkg/deploy/pipeline.go

```diff
@@ -160,10 +161,15 @@ func PipelineSourceFromManifest(mfSource *manifest.Source) (source interface{},
 			return repo, false, nil
 		}
 	case manifest.CodeCommitProviderName:
+		outputFormat, ok := mfSource.Properties["output_artifact_format"]
+		if !ok {
+			outputFormat = ""
+		}
 		return &CodeCommitSource{
-			ProviderName:  manifest.CodeCommitProviderName,
-			Branch:        (mfSource.Properties["branch"]).(string),
-			RepositoryURL: (mfSource.Properties["repository"]).(string),
+			ProviderName:         manifest.CodeCommitProviderName,
+			Branch:               (mfSource.Properties["branch"]).(string),
+			RepositoryURL:        (mfSource.Properties["repository"]).(string),
+			OutputArtifactFormat: (outputFormat).(string),
```

Comment:
Maybe just
```suggestion
			OutputArtifactFormat: (mfSource.Properties["branch"]).(string),
```
since if the key doesn't exist, `mfSource.Properties["branch"]` would still be an empty string.

Reply:
no `Properties` is a `map[string]interface{}`. It panics if we convert a `nil` `interface{}` to `string`

Reply:
Oooh right i guess branch and repository are all required fields? But does that mean if users remove them from manifest we'll panic as well?

Reply:
I think so!! Let me take care of it.